From 5bd7e357b46b3a2fb0b9131b214a2466d047c6e9 Mon Sep 17 00:00:00 2001
From: Xiaowen Liu <b37945@freescale.com>
Date: Tue, 11 Dec 2012 10:43:35 +0800
Subject: [PATCH 01/43] ENGR00215174 add GRALLOC_USAGE_HW_FBX usage for
 FramebufferSurface.

add GRALLOC_USAGE_HW_FBX usage for FramebufferSurface to indicate
second display.

Signed-off-by: Xiaowen Liu <b37945@freescale.com>
---
 services/surfaceflinger/DisplayHardware/FramebufferSurface.cpp | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/services/surfaceflinger/DisplayHardware/FramebufferSurface.cpp b/services/surfaceflinger/DisplayHardware/FramebufferSurface.cpp
index 6ef3295..55f215b 100644
--- a/services/surfaceflinger/DisplayHardware/FramebufferSurface.cpp
+++ b/services/surfaceflinger/DisplayHardware/FramebufferSurface.cpp
@@ -61,9 +61,12 @@ FramebufferSurface::FramebufferSurface(HWComposer& hwc, int disp,
 {
     mName = "FramebufferSurface";
     mConsumer->setConsumerName(mName);
-    mConsumer->setConsumerUsageBits(GRALLOC_USAGE_HW_FB |
-                                       GRALLOC_USAGE_HW_RENDER |
-                                       GRALLOC_USAGE_HW_COMPOSER);
+    int usage = GRALLOC_USAGE_HW_FB | GRALLOC_USAGE_HW_RENDER |
+                                       GRALLOC_USAGE_HW_COMPOSER;
+    if(disp != 0) {
+        usage |= GRALLOC_USAGE_HW_FBX;
+    }
+    mConsumer->setConsumerUsageBits(usage);
     mConsumer->setDefaultBufferFormat(mHwc.getFormat(disp));
     mConsumer->setDefaultBufferSize(mHwc.getWidth(disp),  mHwc.getHeight(disp));
     mConsumer->setDefaultMaxBufferCount(NUM_FRAMEBUFFER_SURFACE_BUFFERS);
-- 
1.8.0

